shared/js/background/events/3p-tracking-cookie-blocking.js

```diff
@@ -3,6 +3,21 @@ const trackerutils = require('../tracker-utils')
 const utils = require('../utils.es6')
 const devtools = require('../devtools.es6')
 
+function shouldBlockHeaders (request, tab, requestIsTracker) {
+    if (requestIsTracker && !tab.site.isFeatureEnabled('trackingCookies3p')) {
+        return false
+    }
+    if (!requestIsTracker && !tab.site.isFeatureEnabled('nonTracking3pCookies')) {
+        return false
+    }
+
+    if (trackerutils.isFirstPartyByEntity(request.url, tab.url)) {
```

Comment:
@sammacbeth + @SlayterDev do we know if this is infallible, I'd prefer for breakage sake this function failed safe if we're uncertain.

Reply:
In the case where we're unsure (i.e. we can't find an entity for either domain) they're compared by the domain itself.

Reply:
Cool that sounds ok then